client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -440,14 +444,32 @@ export class PlanFeatures2023Grid extends Component< PlanFeatures2023GridType >
 	}
 
 	handleUpgradeClick = ( planSlug: PlanSlug ) => {
-		const { onUpgradeClick: ownPropsOnUpgradeClick, gridPlansForFeaturesGrid } = this.props;
-		const { cartItemForPlan } =
+		const {
+			onUpgradeClick: ownPropsOnUpgradeClick,
+			gridPlansForFeaturesGrid,
+			selectedStorageOptions,
+		} = this.props;
+		const { cartItemForPlan, storageAddOnsForPlan } =
 			gridPlansForFeaturesGrid.find( ( gridPlan ) => gridPlan.planSlug === planSlug ) ?? {};
 
-		// TODO clk: Revisit. Could this suffice: `ownPropsOnUpgradeClick?.( cartItemForPlan )`
+		const selectedStorageOption =
+			selectedStorageOptions && selectedStorageOptions[ cartItemForPlan?.product_slug || '' ];
```

Comment:
In general, I think it's best if we don't retrieve things from data structures using fallback empty values like this. It confuses further reasoning with the type. In this case, `selectedStorageOption` is always defined despite `''` being used as the effective value.

However, this can be simplified to:

```
const selectedStorageOption = selectedStorageOptions?.[ planSlug ];
```

Otherwise, maybe consider a more deterministic expression?:

```
const selectedStorageOption = cartItemForPlan?.product_slug ? selectedStorageOptions?.[ cartItemForPlan?.product_slug ] : undefined;
```

Reply:
Thank you! Makes total sense. I'll roll this into the other code cleanup I wanted to wrap up this morning 👍

Reply:
Tidied up your suggestion and a subsequent empty value comparison https://github.com/Automattic/wp-calypso/pull/81340/commits/456b5d64d4fff73aa1e48e3a21f46fc0c4881c4f